app/managers/security_manager/index.ts

```diff
@@ -0,0 +1,477 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import Emm from '@mattermost/react-native-emm';
+import {isRootedExperimentalAsync} from 'expo-device';
+import {createIntl, defineMessages} from 'react-intl';
+import {Alert, type AlertButton, AppState, type AppStateStatus, Platform} from 'react-native';
+
+import {switchToServer} from '@actions/app/server';
+import {logout} from '@actions/remote/session';
+import {Preferences} from '@constants';
+import {DEFAULT_LOCALE, getTranslations, t} from '@i18n';
+import {getServerCredentials} from '@init/credentials';
+import ManagedApp from '@init/managed_app';
+import {toMilliseconds} from '@utils/datetime';
+import {isMainActivity} from '@utils/helpers';
+import {logError} from '@utils/log';
+
+import type {AvailableScreens} from '@typings/screens/navigation';
+
+type SecurityManagerServerConfig = {
+    Biometrics?: boolean;
+    JailbreakProtection?: boolean;
+    PreventScreenCapture?: boolean;
+    authenticated?: boolean;
+    lastAccessed?: number;
+    siteName?: string;
+};
+
+type SecurityManagerServersCollection = Record<string, SecurityManagerServerConfig>;
+
+class SecurityManagerSingleton {
+    activeServer?: string;
+    serverConfig: SecurityManagerServersCollection = {};
+    backgroundSince = 0;
+    previousAppState?: AppStateStatus;
+    initialized = false;
+
+    constructor() {
+        AppState.addEventListener('change', this.onAppStateChange);
+    }
+
+    /**
+     * Initializes the class with existing servers on app launch.
+     * Should be called when the app starts.
+     */
+    async init(servers: Record<string, ClientConfig>, activeServer?: string) {
+        if (this.initialized) {
+            return;
+        }
+
+        this.initialized = true;
+        const added = new Set<string>();
+        for (const [server, config] of Object.entries(servers)) {
+            if (!this.serverConfig[server]) {
+                this.addServer(server, config);
+                added.add(server);
+            }
+        }
+
+        if (activeServer && (!added.has(activeServer) || !this.activeServer)) {
+            this.activeServer = activeServer;
+            this.setScreenCapturePolicy(activeServer);
+            const isJailbroken = await this.isDeviceJailbroken(activeServer);
+            if (!isJailbroken) {
+                this.authenticateWithBiometricsIfNeeded(activeServer);
+            }
+        }
+    }
+
+    /**
+     * Handles app state changes to prompt authentication when resuming from background.
+     */
+    onAppStateChange = async (appState: AppStateStatus) => {
+        if (this.isAuthenticationHandledByEmm()) {
+            return;
+        }
+
+        const isActive = appState === 'active';
```

Comment:
nit - create constants for these states.

Reply:
no need, this are protected by typescript set by `AppStateStatus` which is defined as `type AppStateStatus = "active" | "background" | "inactive" | "unknown" | "extension"`